skyportal/models/photometry.py

```diff
@@ -201,6 +214,145 @@ def e_mag(cls):
             else_=None,
         )
 
+    @property
+    def magref(self):
+        """
+        Reference magnitude, e.g.,
+        the mean magnitude of a variable source,
+        or the magnitude of a source before a transient started.
+        """
+        if self.ref_flux is not None and self.ref_flux > 0:
+            return -2.5 * np.log10(self.ref_flux) + PHOT_ZP
+        else:
+            return None
+
+    @hybrid_property
+    def e_magref(self):
+        """The error on the reference magnitude."""
+        if (
+            self.ref_flux is not None
+            and self.ref_flux > 0
+            and self.ref_fluxerr is not None
+            and self.ref_fluxerr > 0
+        ):
+            return (2.5 / np.log(10)) * (self.ref_fluxerr / self.ref_flux)
+        else:
+            return None
+
+    @magref.expression
+    def magref(cls):
+        """The magnitude of the photometry point in the AB system."""
+        return sa.case(
+            [
+                (
+                    sa.and_(
+                        cls.ref_flux != None,  # noqa: E711
+                        cls.ref_flux != 'NaN',
+                        cls.ref_flux > 0,
+                    ),
+                    -2.5 * sa.func.log(cls.ref_flux) + PHOT_ZP,
+                )
+            ],
+            else_=None,
+        )
+
+    @e_magref.expression
+    def e_magref(cls):
+        """The error on the magnitude of the photometry point."""
+        return sa.case(
+            [
+                (
+                    sa.and_(
+                        cls.ref_flux != None,  # noqa: E711
+                        cls.ref_flux != 'NaN',
+                        cls.ref_flux > 0,
+                        cls.ref_fluxerr > 0,
+                    ),  # noqa: E711
+                    2.5 / sa.func.ln(10) * cls.ref_fluxerr / cls.ref_flux,
```

Comment:
Can you put parentheses in this equation to make sure the order of operations is what you expect? I'm not a far of double division without parentheses (even if not needed, useful for readability)

Reply:
will do